### PR TITLE
`Programming exercises`: Show warning in build plan editor if build plan could not be found

### LINKED ISSUE #6784: `Programming exercises`: Build plan editor page remains empty for exercises created before 6.3.0
### Describe the bug

Artemis stores the build plans for Jenkins setups in the database and allows editing them in the UI.

For exercises created before that change the text box remains empty.

### To Reproduce

1. Artemis with GitLab/Jenkins setup. A programming exercise created before update 6.3.0.
2. On the programming exercise details page, go to ‘Edit Build Plan’.
3. The text box remains empty.

### Expected behavior

Some information message and/or warning should be shown to the user that this feature is not available for this exercise.

Additional relevant information to the user would be that using the ‘Reset’ functionality to reset the build plans would result in the creation of new build plans from the template and then allow customising them from there in the UI even for such older exercises.

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

≧ 6.3.0

### What browsers are you seeing the problem on?

Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [x] ~~I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).~~ Not applicable, feature not available there.
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context
Closes #6784

### Description
Handles the error sent from the server and shows a warning to the user.

### Steps for Testing

Prerequisites:
- Gitlab/Jenkins setup (e.g. TS2)
- 1 Instructor
- 1 old Programming Exercise *created before 6.3.0*

1. Go the the programming exercise details page in the course management.
2. Button block at the top: → Edit build plan
3. A warning should be shown.
4. Create a new programming exercise.
5. Go to the Edit build plan again, this time no warning is shown and the editor is filled with content.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| `build-plan-editor.component.ts` | 98% | ✅ |

### Screenshots
n/a

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts [hunk 1]
```diff
@@ -67,10 +74,16 @@ export class BuildPlanEditorComponent implements AfterViewInit, OnInit {
...
-            this.initEditor();
-            this.isLoading = false;
+        this.buildPlanService.getBuildPlan(exerciseId).subscribe({
+            next: (buildPlan) => {
+                this.buildPlan = buildPlan.body!;
+                this.initEditor();
+                this.isLoading = false;
+            },
+            error: () => {
+                this.alertService.error('artemisApp.programmingExercise.buildPlanFetchError');
```
Comment (b-fein) [inline src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts:84]: ToDo: test case that alertservice was called on error and loading=false afterwards

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 2]
```diff
@@ -98,6 +98,7 @@
             "buildplans": "Build-Pläne",
             "buildplan": "Build-Plan",
             "buildPlanEditor": "Build-Plan-Editor",
+            "buildPlanFetchError": "Konnte keinen Build-Plan für diese Aufgabe finden. Setze die Build-Pläne zurück, um einen Build-Plan zu erstellen.",
```
Comment (b-fein) [inline src/main/webapp/i18n/de/programmingExercise.json:101]: ToDo: mention old exercise format with non-editable build plans in message?

**Review by MaximilianAnzinger — APPROVED** [review]
Code LGTM

**Review by DominikRemo — DISMISSED** [review]
Code LGTM

**Review by chrisknedl — DISMISSED** [review]
Tested this on my local setup. I created a programming exercise on version 6.2.2 and everything worked exactly as described in the testing steps.

**Review by reschandreas — APPROVED** [review]
Tested on TS2. Works as expected. Thanks 👍🏻

**Comment by b-fein** [comment]
Merge (e8b0a767b3aa97d0df7d393213dc3c1833a35d0c) only changed some minor formatting due to the linter dependency update. No actual code changes, and no full re-review necessary.

File: src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts [hunk 3]
```diff
@@ -71,10 +73,17 @@ export class BuildPlanEditorComponent implements AfterViewInit, OnInit {
...
-            this.buildPlan = buildPlan.body!;
-            this.initEditor();
-            this.isLoading = false;
+        this.buildPlanService.getBuildPlan(exerciseId).subscribe({
+            next: (buildPlan) => {
+                this.buildPlan = buildPlan.body!;
+                this.initEditor();
+                this.isLoading = false;
+            },
+            error: () => {
```
Comment (bassner) [inline src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts:82]: I think we should actually inspect the error here and only display this alert for the exact expected error (404?). There could be other error types here in exceptional cases, right?

File: src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts [hunk 4]
```diff
@@ -52,7 +54,7 @@ export class BuildPlanEditorComponent implements AfterViewInit, OnInit {
     ngAfterViewInit(): void {
         this.editor.getEditor().setOptions({
             animatedScroll: true,
-            maxLines: 80,
+            maxLines: 45,
```
Comment (bassner) [inline src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts:57]: How is this change related to the PR? :)

**Review by bassner — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts (hunk 4 above)
Comment (b-fein) [reply src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts:57]: While working on this page I noticed that even in full-HD-fullscreen the box was extending too far. This was the easiest fix for that without having to dig into how the editor box can be dynamically resized (IIRC the online code editor also does not resize automatically, but in this case some manual dragging is possible to manually change the size, could be done as a separate PR in the future).

File: src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts (hunk 3 above)
Comment (b-fein) [reply src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts]: Only for 404 errors the specific message is shown now, for other errors the generic global error handler is called b51300e284ac96088e40aa436bedfd26629fa5f0.

**Review by bassner — APPROVED** [review]
Thanks for adding the error inspection. LGTM 👍

**Review by chrisknedl — APPROVED** [review]
Re-approve

**Review by reschandreas — APPROVED** [review]
reapprove

**Pull request merged**